### PR TITLE
SW-5000 Implement deliverable status update

### PR DESCRIPTION
**Pull request by sgrimm**
Replace the stub implementation of the deliverable status update endpoint with a
real one.

### CODE REVIEW

**Comment by sgrimm** [comment]
* **#1731** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1731?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#1730** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1730?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1729** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1729?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @sgrimm and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/test/kotlin/com/terraformation/backend/accelerator/db/SubmissionStoreTest.kt [hunk 1]
```diff
@@ -64,4 +71,108 @@ class SubmissionStoreTest : DatabaseTest(), RunsAsUser {
...
+                  submissionStatusId = SubmissionStatus.InReview)),
+          submissionsDao.findAll())
+    }
+
+    @Test
+    fun `updates existing submission`() {
+      val projectId = insertProject()
+      val deliverableId = insertDeliverable()
+      val feedback = "This looks great"
+      val internalComment = "This made my eyes glaze over"
```
Comment (nickgraz) [inline src/test/kotlin/com/terraformation/backend/accelerator/db/SubmissionStoreTest.kt:109]: 😆

File: src/main/kotlin/com/terraformation/backend/accelerator/db/SubmissionStore.kt [hunk 2]
```diff
@@ -27,6 +33,50 @@ class SubmissionStore(
...
+            .and(SUBMISSIONS.PROJECT_ID.eq(projectId))
+            .fetchOne()
+            ?: SubmissionsRecord(
+                    projectId = projectId,
+                    deliverableId = deliverableId,
+                    submissionStatusId = SubmissionStatus.NotSubmitted,
+                    createdBy = currentUser().userId,
+                    createdTime = now,
+                )
+                .also { it.attach(dslContext.configuration()) }
```
Comment (nickgraz) [inline src/main/kotlin/com/terraformation/backend/accelerator/db/SubmissionStore.kt:60]: So good

**Review by nickgraz — APPROVED** [review]
Nice work, thanks for the quick turnaround

**Comment by sgrimm** [comment]
### Merge activity

* **Mar 8, 12:49 PM EST**: @sgrimm started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/1731).
* **Mar 8, 12:54 PM EST**: [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/1731) rebased this pull request as part of a merge.
* **Mar 8, 12:55 PM EST**: @sgrimm merged this pull request with [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/1731).

**Pull request merged**